colossalai/booster/booster.py

```diff
@@ -64,17 +65,20 @@ def __init__(self,
             assert isinstance(
                 plugin, Plugin), f'Expected the argument plugin to be an instance of Plugin, but got {type(plugin)}.'
         self.plugin = plugin
+        logger = get_dist_logger()
 
         # set accelerator
         if self.plugin and self.plugin.control_device():
             self.accelerator = None
-            warnings.warn('The plugin will control the accelerator, so the device argument will be ignored.')
+            logger.warning('The plugin will control the accelerator, so the device argument will be ignored.',
```

Comment:
Can we set default value of `device`  arg of this function to `None`? Then we can only warn if `device` is not None here. And if no plugin is used or plugin does not control device, we can set `device = device or 'cuda'`. In the docstring, we have to declare the default value is cuda. @FrankLeeeee

Reply:
modified